### PR TITLE
Make lodash memoize retain type information

### PR DESCRIPTION
**Pull request by aaronjensen**
Looks like prettier did its thing on the test file, but the new tests are at the bottom.

### CODE REVIEW

File: definitions/npm/lodash_v4.x.x/flow_v0.47.x-/test_lodash-v4.x.x.js [hunk 1]
```diff
@@ -329,43 +321,58 @@ directStrings = _.map(nums, function(num) {
...
 // $ExpectError string. This type is incompatible with number
-var strings : string[] = _.times(5);
-timesNums = _.times(5, function(i: number) { return i + 1; });
+var strings: string[] = _.times(5);
+timesNums = _.times(5, function(i: number) {
+  return i + 1;
+});
 // $ExpectError string. This type is incompatible with number
-timesNums = _.times(5, function(i: number) { return JSON.stringify(i); });
+timesNums = _.times(5, function(i: number) {
```
Comment (GAntoine) [inline definitions/npm/lodash_v4.x.x/flow_v0.47.x-/test_lodash-v4.x.x.js]: Try putting this back on a single line, it might fix the build error.

Comment (aaronjensen) [reply definitions/npm/lodash_v4.x.x/flow_v0.47.x-/test_lodash-v4.x.x.js]: Thanks for pointing this out, I missed the failure. Putting it on one line isn't an option due to prettier, but I did move the comment to the right place.

**Comment by aaronjensen** [comment]
looks like this stuff all changed out from under me. I'm trying to get it updated, but for some reason I can't get $ExpectError tests to fail anymore.

**Comment by aaronjensen** [comment]
Ok, looks like this may be good now, let's see if it passes CI

**Comment by aaronjensen** [comment]
Am I doing something wrong that is causing these to get run through prettier?

**Comment by GAntoine** [comment]
Looks like CI passed this time.

> Am I doing something wrong that is causing these to get run through prettier?

Are you saying that you're not running prettier locally, and that the changes are happening after you open the PR?

**Comment by aaronjensen** [comment]
I believe it is running in a precommit hook?

**Comment by GAntoine** [comment]
Hmm, there doesn't seem to be a post-commit hook in the project, so I'm assuming your system is setup to run it. Regardless, I don't see a problem with prettification, as long as CI is happy and the changes are valid.

**Comment by aaronjensen** [comment]
It does via `lint-staged` afaict

**Comment by aaronjensen** [comment]
https://github.com/flowtype/flow-typed/blob/498ec718356ce17cdcbf9ddbf2e8efd522a78fd1/definitions/package.json#L9

**Comment by aaronjensen** [comment]
ugh. Did not mean to close.

**Comment by GAntoine** [comment]
Whoops, I totally missed that. Well I'm no flow pro, but your change seems to match the lodash API, so, approved!

**Pull request merged**